src/devices/MemoryLCD/LSxxxB7DHxx.cs

```diff
@@ -0,0 +1,208 @@
+        {
+            if (data.Length != lineIndex.Length * BytesPerLine)
+            {
+                throw new Exception($"Data writing period must be {lineIndex.Length * PixelWidth}ck ({lineIndex.Length * BytesPerLine} bytes)");
+            }
+
+            // m(1), ag1(1), d1(18), dummy2(1), ag2(1), d2(18), ... , dummyn(1), agn(1), dn(18), dummy(2)
+            var buffer = new byte[2 + (2 + BytesPerLine) * lineIndex.Length];
+
+            buffer[0] = (byte)(ModeSelectionPeriodByte.Mode | (frameInversion ? ModeSelectionPeriodByte.FrameInversion : ModeSelectionPeriodByte.Dummy));
+
+            for (var i = 0; i < lineIndex.Length; i++)
+            {
+                var from = BytesPerLine * i;
+                var to = 1 + (2 + BytesPerLine) * i;
+
+                buffer[to] = Utility.GetAgByte(lineIndex[i] + 1);
+
+                Span<byte> fromLine = data.Slice(from, BytesPerLine);
+                Span<byte> toLine = buffer.AsSpan(to + 1, BytesPerLine);
+
+                fromLine.CopyTo(toLine);
+            }
+
+            WriteSpi(buffer);
+        }
+
+        /// <summary>
+        /// Maintains memory internal data (maintains current display).
+        /// </summary>
+        public void Display(bool frameInversion = false)
+        {
+            // m(1), dummy(1)
+            var buffer = new byte[2];
+
+            buffer[0] = (byte)(frameInversion ? ModeSelectionPeriodByte.FrameInversion : ModeSelectionPeriodByte.Dummy);
+
+            WriteSpi(buffer);
+        }
+
+        /// <summary>
+        /// Clears memory internal data and writes white.
+        /// </summary>
+        public void AllClear(bool frameInversion = false)
+        {
+            // m(1), dummy(1)
+            var buffer = new byte[2];
+
+            buffer[0] = (byte)(ModeSelectionPeriodByte.AllClear | (frameInversion ? ModeSelectionPeriodByte.FrameInversion : ModeSelectionPeriodByte.Dummy));
+
+            WriteSpi(buffer);
+        }
+
+        private void Init()
+        {
+            if (_gpio != null)
+            {
+                if (_scs > -1)
+                {
+                    _gpio.OpenPin(_scs, PinMode.Output);
+                }
+
+                if (_disp > -1)
+                {
+                    _gpio.OpenPin(_disp, PinMode.Output);
+                    _gpio.Write(_disp, PinValue.High);
+                    Thread.Sleep(s_powerup_disp_delay);
+                }
+
+                if (_extcomin > -1)
+                {
+                    _gpio.OpenPin(_extcomin, PinMode.Output);
+                    _gpio.Write(_extcomin, PinValue.Low);
+                    Thread.Sleep(s_powerup_extcomin_delay);
+                }
+            }
+        }
+
+        private void WriteSpi(ReadOnlySpan<byte> bytes)
+        {
+            if (_gpio != null && _scs > -1)
+            {
+                _gpio.Write(_scs, PinValue.High);
```

Comment:
IIRC Chip select lines are usually low active. Is it really correct to set it high here?

Reply:
yes for sure, see datasheet section 6. all 3 model had been tested on my device

IMO such the timing may facilitate compatibility with other SPI devices during transmission